### DISCUSSION TITLE
重跑工作流，业务日期${system.datetime} 没有使用原来的？？

### DISCUSSION THREAD

**Discussion by momomobinx** [apache/dolphinscheduler#9179] [Q&A]
我重新执行一个工作流，工作流中使用了一个$ {system.datetime}  全局变量，这个全局变量不是 以前执行的时间，使用 $[yyyyMMddHHmmss]也是，看了下代码，好像只有定时调度执行的 任务实例 才有调度时间，然后重跑会 使用原来的调度时间。如果是手动点开始执行的 那么任务实例就没有调度时间 ，然后在参数替换的时候就会使用一个新的时间去替换 $[yyyyMMddHHmmss] 变量

**Comment by Tandoy** [comment]
Yes, it is the timing time of the daily scheduling instance

https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/parameter/built-in.html

**Reply by momomobinx** [reply]
重跑不是应该让工作流像是还是 过去的时间执行的一样吗？ 比如与时间见相关的 任务

**Reply by momomobinx** [reply]
也就是在工作流实例执行的时候 ，相关的参数应该是定下来了，然后再次执行，参数还是一样的

**Reply by Tandoy** [reply]
${system.datetime} This is to get the current time of the system. If you need the time to remain unchanged, you can try $[yyyymmdd-1]. It's just my understanding.

**Reply by momomobinx** [reply]
我试试看

**Reply by momomobinx** [reply]
> 

不行，工作流实例还是以前的，但是任务实例是新的，使用的新的时间

**Reply by Tandoy** [reply]
![企业微信截图_20220325115538](https://user-images.githubusercontent.com/56899730/160051333-07e66f9b-8c45-48e2-b08d-2fd2d53853da.png)

![企业微信截图_20220325115504](https://user-images.githubusercontent.com/56899730/160051347-d2cd23fe-c5cd-4f6c-87d6-83f8ac2f32b0.png)

![企业微信截图_20220325115521](https://user-images.githubusercontent.com/56899730/160051359-ddd7cade-33a1-4155-8747-c1356423768d.png)

**Reply by momomobinx** [reply]
这个是减一天啊，重跑还是当前日期减一天，20220323 的业务日期 按照这个重跑 日期就是 20220324

**Reply by Tandoy** [reply]
> 这个是减一天啊，重跑还是当前日期减一天，20220323 的业务日期 按照这个重跑 日期就是 20220324

The above example: I selected the complement scheduling 20220323 in 20220325, and even if I rerun, it will also be 20220323. Isn't that right? Maybe you can configure the workflow like this and try it

**Reply by momomobinx** [reply]
https://github.com/apache/dolphinscheduler/issues/9150#issuecomment-1078720670
我创建两个http节点然后又遇到了这个错误

**Reply by momomobinx** [reply]
> 

我的是手动触发的，调度时间那里是“-”,然后点击重跑就是新的时间

**Reply by momomobinx** [reply]
> > 这个是减一天啊，重跑还是当前日期减一天，20220323 的业务日期 按照这个重跑 日期就是 20220324
> 
> The above example: I selected the complement scheduling 20220323 in 20220325, and even if I rerun, it will also be 20220323. Isn't that right? Maybe you can configure the workflow like this and try it

一直都是新的时间啊 $[yyyyMMddHHmmss]  你的是什么版本啊

**Reply by momomobinx** [reply]
> > 这个是减一天啊，重跑还是当前日期减一天，20220323 的业务日期 按照这个重跑 日期就是 20220324
> 
> The above example: I selected the complement scheduling 20220323 in 20220325, and even if I rerun, it will also be 20220323. Isn't that right? Maybe you can configure the workflow like this and try it

只有定时调度的任务在恢复失败后才会使用原来是的时间，如果是手动调度的，那么就恢复失败就会使用新的时间
